onnxruntime/python/tools/quantization/E2E_example_model/nlp/bert/trt/e2e_tensorrt_bert_example.py

```diff
@@ -0,0 +1,129 @@
+        if iter_data:
+            return iter_data
+
+        self.enum_data_dicts = None
+        if self.current_index + self.batch_size > self.num_inputs:
+            print("Calibrating index {:} batch size {:} exceed max input limit {:} sentences".format(self.current_index, self.batch_size, self.num_inputs))
+            return None
+
+        current_batch = int(self.current_index / self.batch_size)
+        if current_batch % 10 == 0:
+            print("Calibrating batch {:}, containing {:} sentences".format(current_batch, self.batch_size))
+
+        input_ids = []
+        input_mask = []
+        segment_ids = []
+        for i in range(self.batch_size):
+            example = self.data[self.current_index + i]
+            features = dp.convert_example_to_features(example.doc_tokens, example.question_text, self.tokenizer, self.max_seq_length, self.doc_stride, self.max_query_length)
+            self.example_id_list.append(example.id)
+            self.feature_list.append(features)
+            self.token_list.append(example.doc_tokens)
+            if len(input_ids) and len(segment_ids) and len(input_mask):
+                input_ids = np.vstack([input_ids, features[0].input_ids])
+                input_mask = np.vstack([input_mask, features[0].input_mask])
+                segment_ids = np.vstack([segment_ids, features[0].segment_ids])
+            else:
+                input_ids = np.expand_dims(features[0].input_ids, axis=0)
+                input_mask = np.expand_dims(features[0].input_mask, axis=0)
+                segment_ids = np.expand_dims(features[0].segment_ids, axis=0)
+        data = [{"input_ids": input_ids, "input_mask": input_mask, "segment_ids":segment_ids}]
+        print(input_ids.shape)
+        self.current_index += self.batch_size
+        self.enum_data_dicts = iter(data)
+        return next(self.enum_data_dicts, None)
+
+if __name__ == '__main__':
+    '''
+    BERT QDQ Quantization for TensorRT.
+
+    There are two steps for the quantization,
+    first, calibration is done based on SQuAD dataset to get dynamic range of each floating point tensor in the model
+    second, Q/DQ nodes with dynamic range (scale and zero-point) are inserted to the model
+
+    The onnx model used in the script is converted from Hugging Face BERT model,
+    https://huggingface.co/transformers/serialization.html#converting-an-onnx-model-using-the-transformers-onnx-package
+
+    Some utility functions for dataset processing and data reader are from Nvidia TensorRT demo BERT repo,
+    https://github.com/NVIDIA/TensorRT/tree/master/demo/BERT
+    '''
+
+    # Model and Dataset settings
+    model_path = "./model.onnx"
+    squad_json = "./dev-v1.1.json"
+    vocab_file = "./vocab.txt"
+    calibration_cache_file = "./bert.flatbuffers"
+    augmented_model_path = "./augmented_model.onnx"
+    qdq_model_path = "./qdq_model.onnx"
+    sequence_lengths = [384]
+    calib_num = 100
+
+    # Generate INT8 calibration cache
```

Comment:
Just call quantizat_static?

Reply:
We also want to save compute range to calibration table for later use in TRT, which is missing in quantize_static.